Add CrowdStrike AIDR guardrail hook

kenany opened the pull request:
## Title

Add CrowdStrike AIDR guardrail hook

## Pre-Submission checklist

**Please complete all items before asking a LiteLLM maintainer to review your PR**

- [x] I have Added testing in the [`tests/litellm/`](https://github.com/BerriAI/litellm/tree/main/tests/litellm) directory, **Adding at least 1 test is a hard requirement** - [see details](https://docs.litellm.ai/docs/extras/contributing_code)
- [x] I have added a screenshot of my new test passing locally 
- [x] My PR passes all unit tests on [`make test-unit`](https://docs.litellm.ai/docs/extras/contributing_code)
- [x] My PR's scope is as isolated as possible, it only solves 1 specific problem

<img width="1388" height="415" alt="2025-12-12-14-35-45" src="https://github.com/user-attachments/assets/17ff5453-08ec-441c-9ba9-298c1dd58f76" />

## Type

<!-- Select the type of Pull Request -->
<!-- Keep only the necessary ones -->

🆕 New Feature

## Changes

Functionally similar to the existing Pangea AI Guard guardrail hook, just under the CrowdStrike umbrella now.

kenany committed:
Add CrowdStrike AIDR guardrail hook

krrishdholakia commented on litellm/proxy/guardrails/guardrail_hooks/crowdstrike_aidr/crowdstrike_aidr.py:
```diff
@@ -0,0 +1,328 @@
+
+        return result
+
+    async def _async_pre_call_hook(
```
@kenany can we please just use the `apply_guardrails` event hook **instead** of async_pre and async_post call

this will allow us to guarantee your guardrail works across all our llm endpoints (chat completions, responses api, anthropic messages, etc.)

kenany replied:
Done.

krrishdholakia reviewed (changes requested):
please use the documented `apply_guardrails` endpoint instead of the individual pre/post call hooks, to guarantee this works across multiple llm endpoints. 

If there are any limitations to the event hook, please share them on slack - https://docs.litellm.ai/docs/integrations/community#get-support--connect

kenany committed:
fixup! use apply_guardrail event hook

lapinek commented on tests/test_litellm/proxy/guardrails/guardrail_hooks/test_crowdstrike_aidr.py:
```diff
@@ -0,0 +1,394 @@
+    CrowdStrikeAIDRHandler,
+)
+from litellm.proxy.guardrails.init_guardrails import init_guardrails_v2
+from litellm.types.guardrails import GenericGuardrailAPIInputs
```
GenericGuardrailAPIInputs was moved from litellm.types.guardrails to litellm.types.utils

```suggestion
from litellm.types.utils import GenericGuardrailAPIInputs
```

kenany replied:
Done.

kenany committed:
fixup! update imports

lapinek committed:
fix(guardrails): include AI response in CrowdStrike AIDR output events

Issue:
_build_guard_input_for_response() was:
- Sending only the original user input (messages).
- Not sending the AI provider response.

This fix will:
  - Extract response.choices from the ModelResponse object and include them in guard_input payload.
  - Thus, ensure AIDR output rules receive the AI-generated content for analysis.
  - Fix and update tests.

lapinek committed:
fix(guardrails): prevent duplicate input events in CrowdStrike AIDR guardrail

Issue:
The CrowdStrike AIDR guardrail was running on during_call hooks wihtout event_hook configured.

This fix will:
- Set event_hook to ["pre_call", "post_call"] (AIDR admins will control what policy is applied)

This change will:
- Require default_on parameter
- Prevent duplicate API calls to AIDR for the same input
- Avoid unchecked AI provider API calls on during_call hook

lapinek committed:
docs: add CrowdStrike AIDR to the list of Guardrails under Integrations

lapinek committed:
docs: update CrowdStrike AIDR documentation page
